awx/main/tests/unit/test_tasks.py

```diff
@@ -1472,6 +1522,78 @@ def run_pexpect_side_effect(*args, **kwargs):
         self.run_pexpect.side_effect = run_pexpect_side_effect
         self.task.run(self.pk)
 
+    def test_azure_rm_source_with_tenant(self):
```

Comment:
The branch of code for AzureRM + InventoryUpdate didn't have any test coverage; this adds it.